src/editors/editorsController.ts

```diff
@@ -32,6 +32,21 @@ import TelemetryService from '../telemetry/telemetryService';
 
 const log = createLogger('editors controller');
 
+export function getFileDisplayNameForDocumentId(documentId) {
+  let displayName = EJSON.stringify(documentId);
+
+  // Strip out all special characters to ensure VSCode handles
+  // it correctly in a uri.
+  const regularExpression = /[^a-z0-9{}_-\s":,$!*#@()]/gi;
+  displayName = displayName.replace(regularExpression, '');
```

Comment:
I think we could do a better implementation than this, going to update before merging.